File: api/src/main/java/snowf/urls/api/dto/WebResponse.java
package snowf.urls.api.dto;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
public class WebResponse<T> {
    private MessageHeader header;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    private T data;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    private Errors errors;

    @Builder
    @Data
    @AllArgsConstructor
    @NoArgsConstructor
    public static class MessageHeader {
        private String requestId;
        private String timestamp;
        private String message;
        private String path;
    }

    @Builder
    @Data
    @AllArgsConstructor
    @NoArgsConstructor
    public static class Errors {
        private String errorCode;
        private String errorMessage;

        @JsonInclude(JsonInclude.Include.NON_NULL)
        private Object details;
    }
}

File: web/src/api/dto/web-response.ts
export interface WebResponse<T>{
  messageHeader: MessageHeader;
  data: T;
  Errors: Errors;
}

interface MessageHeader {
  requestId: string;
  timestamp: string;
  message: string;
  path: string;
}

interface Errors {
  errorCode: string;
  errorMessage: string;
  details: Details[];
}

interface Details {
  field: string;
  message: string;
}
